src/main/java/com/rocketden/main/service/ProblemService.java

```diff
@@ -169,21 +173,23 @@ public ProblemDto deleteProblem(String problemId) {
             throw new ApiException(ProblemError.EMPTY_FIELD);
         }
 
+        if (numProblems <= 0) {
+            throw new ApiException(ProblemError.INVALID_NUMBER_REQUEST);
+        }
+
         List<Problem> problems;
         if (difficulty == ProblemDifficulty.RANDOM) {
-            problems = repository.findAll();
+            //Change to findAllByApproval once it works
+            problems = repository.findAllByApproval(true);
         } else {
-            problems = repository.findAllByDifficulty(difficulty);
+            //Change to findAllByDifficultyAndApproval once it works
```

Comment:
nit: remove these comments now that the feature has been implemented

Reply:
k

Reply:
My first though on all the responses lol: [https://www.youtube.com/watch?v=07TinrpKG6o&t=70s](https://www.youtube.com/watch?v=07TinrpKG6o&t=70s).